components/server/ee/src/user/stripe-service.ts

```diff
@@ -35,140 +34,151 @@ export class StripeService {
         return await this.getStripe().setupIntents.create({ usage: "on_session" });
     }
 
-    async findCustomerByUserId(userId: string): Promise<Stripe.Customer | undefined> {
-        return this.findCustomerByQuery(
-            `metadata['${ATTRIBUTION_ID_METADATA_KEY}']:'${AttributionId.render({ kind: "user", userId })}'`,
-        );
+    async findCustomerByUserId(userId: string): Promise<string | undefined> {
+        return this.findCustomerByAttributionId(AttributionId.render({ kind: "user", userId }));
     }
 
-    async findCustomerByTeamId(teamId: string): Promise<Stripe.Customer | undefined> {
-        return this.findCustomerByQuery(
-            `metadata['${ATTRIBUTION_ID_METADATA_KEY}']:'${AttributionId.render({ kind: "team", teamId })}'`,
-        );
+    async findCustomerByTeamId(teamId: string): Promise<string | undefined> {
+        return this.findCustomerByAttributionId(AttributionId.render({ kind: "team", teamId }));
     }
 
-    async findCustomerByQuery(query: string): Promise<Stripe.Customer | undefined> {
+    async findCustomerByAttributionId(attributionId: string): Promise<string | undefined> {
+        const query = `metadata['attributionId']:'${attributionId}'`;
         const result = await this.getStripe().customers.search({ query });
         if (result.data.length > 1) {
             throw new Error(`Found more than one Stripe customer for query '${query}'`);
         }
-        return result.data[0];
+        return result.data[0]?.id;
     }
 
-    async createCustomerForUser(user: User): Promise<Stripe.Customer> {
-        if (await this.findCustomerByUserId(user.id)) {
+    async createCustomerForUser(user: User): Promise<string> {
```

Comment:
I think this service should only deal with attributionIds not users or teams.

Reply:
Agreed. 👍 This can be achieved by factoring out the differences between `createCustomerForUser` and `ForTeam` (e.g. what is the customer name) as customizable arguments of a single method.